### DISCUSSION TITLE
add .minReadySeconds parameter to helm chart

### DISCUSSION THREAD

**Discussion by alexku7** [jertel/elastalert2#1242] [Ideas]
Since the pod doesn't expose today any liveness endpoints, it's impossible to detect whether  the pod is healthy and ready.

We can mitigate this issue by adding `minReadySeconds` tot the pod specs https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#min-ready-seconds.

So at least we can wait X seconds in order to ensure that the pod is not failing after helm deployment.

**Comment by jertel** [comment]
After the minReadySeconds elapses, how then will the pod failure detection work?

**Reply by alexku7** [reply]
if the pod has not been failed, the pod is considered as healthy. The helm exits with success code.

If the pod has failed after few seconds ( like it happened in my case due the misconfiguration) the helm will not exit with success code as long the pod can't reach the minReadySeconds period without failing. Eventually the helm will reach a timeout and roll back the failing misconfigured deployment.

Today , since the pod doesn't expose any liveness probes it's considered as ready and healthy immediately after starting which causes the helm to exit with success code although the pod might fail few seconds later during the config loading

**Reply by jertel** [reply]
I see. Thanks for the explanation. Feel free to submit a PR with that setting. The chart README should be updated to show the new setting and its default value.
